### ISSUE TITLE
idea: anaphora 

### ISSUE THREAD

**Issue by Luis-Henriquez-Perez** [okamsn/loopy#23]
The idea is this. Sometimes I want to use an unnamed `collect` throughout loopy. But in the `after-do` or `else-do` clause I want to reference the list collect _just one time_. This would be even more annoying the more collects I used (this is assuming, however, that collects will accumulate into the same variable #21) I often lament having to named the list just to reference it one time at the end. Perhaps an anaphoric variable could be of use for this case. Maybe `$result` or `<result>`? Don't know.

**Comment by okamsn** [comment]
I have implemented this and #21 in the latest changes. Please try the out.

I went with the name `loopy-result` for unnamed loops and `loopy-NAME-result` for named loops. Because of how the code is constructed, changing the variable in `after-do` will work as expected but changing it in `finally-do` alone won't work as expected, because at that point the return value of the `cl-block` has already been captured by a `prog1` form.  However, you can still return this changed value with the `finally-return` macro argument, as expected.

```el
(ert-deftest implicit-accumulation-name ()
  (should
   (and (equal '(1 2 3)
               (eval (quote (loopy ((list i '(1 2 3))
                                    (collect i))
                                   (else-do (cl-return loopy-result))))))
        (equal '(0 1 2 3)
               (eval (quote (loopy ((list i '(1 2 3))
                                    (collect i))
                                   (else-do
                                    (push 0 loopy-result)
                                    (cl-return loopy-result))))))
        (equal '(0 1 2 3)
               (eval (quote (loopy ((list i '(1 2 3))
                                    (collect i))
                                   (finally-do
                                    (push 0 loopy-result))
                                   (finally-return loopy-result)))))
        (equal '(1 2 3)
               (eval (quote (loopy ((list i '(1 2 3))
                                    (collect i))
                                   (return loopy-result)))))
        (equal '(1 2 3)
               (eval (quote (loopy my-loop
                                   ((list i '(1 2 3))
                                    (collect i))
                                   (return loopy-my-loop-result))))))))
```

**Comment by okamsn** [comment]
UPDATE: The variable is no longer based on the loop name.

Now that the `loop` macro argument has been replaced by a `sub-loop` or `loop`
macro argument, and now that there is a `without` macro argument, it is not as
useful to keep this behavior.

Having all loops within the `loopy` call accumulate into the same `loopy-result`
makes working with sub-loops easier.

``` emacs-lisp
(loopy outer
       (list i '((1 2) (3 'bad-val) (4 5)))
       (loop (list j i)
             (if (not (numberp j))
                 (return-from outer j)
               (collect j))))
```
